### PR TITLE
Add FileUtils#cp

### PR DESCRIPTION
**Pull request by Dreauw**
Adds a method to copy a file or a list of files in the class FileUtils.
And the associated tests in spec/std/file_utils_spec.cr

:dolphin:

### CODE REVIEW

File: src/file_utils.cr [hunk 1]
```diff
@@ -24,4 +24,29 @@ module FileUtils
       return true if read1 == 0
     end
   end
+
+  # Copies a file content *src* to *dest*. If dest is a directory, copies *src* to *dest/src*
```
Comment (jhass) [inline src/file_utils.cr]: `file contents of *src_path*` might be less confusing. Also mentioning that the basename is taken in the second case might be worth it.

**Comment by Dreauw** [comment]
Yep, the documentation was confusing so I rewrote it.

Should be better now :)

File: src/file_utils.cr [hunk 2]
```diff
@@ -24,4 +24,31 @@ module FileUtils
...
+
+  # Copies a list of files *src* to *dest*.
+  # *dest* must be a directory.
+  # ```
+  # FileUtils.cp({"cgi.cr", "complex.cr", "date.cr"}, "files")
+  # ```
+  def cp(srcs : Enumerable(String), dest : String)
+    Dir.mkdir(dest) rescue nil
+    srcs.each do |src|
+      cp(src, dest + File::SEPARATOR)
```
Comment (jhass) [inline src/file_utils.cr]: Actually why append `File::SEPARATOR`? I think `raise ArgumentError.new("...") unless Dir.exists?(dest)` would be better :)

**Comment by Dreauw** [comment]
Yep, the "File::SEPARATOR" is totally useless, good catch !

Good idea about raising an error, I dunno why I didn't thought of that in the first place (and it's the behavior used in ruby actually).
I changed the code and added a test case :)

**Comment by jhass** [comment]
LGTM :+1:

**Comment by kirbyfan64** [comment]
Could you make this also copy the basic attributes (e.g. mode) like [Python's `shutil.copy`](https://docs.python.org/3/library/shutil.html#shutil.copy) does?

**Comment by Dreauw** [comment]
@kirbyfan64 That should be already the case. The permissions bits are copied too.

**Comment by ysbaddaden** [comment]
Thanks! FileUtils is desperately in need for more :heart:

**Pull request merged**